client-tools/Shell/V8ClientConnection.cpp

```diff
@@ -1012,6 +1010,132 @@ static void ClientConnection_httpPostRaw(
 
 #ifdef ARANGODB_ENABLE_FAILURE_TESTS
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief ClientConnection method "startTelemetrics"
+////////////////////////////////////////////////////////////////////////////////
+
+static void ClientConnection_startTelemetrics(
+    v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+  if (isExecutionDeadlineReached(isolate)) {
+    return;
+  }
+
+  // get the connection
+  V8ClientConnection* v8connection = TRI_UnwrapClass<V8ClientConnection>(
+      args.Holder(), WRAP_TYPE_CONNECTION, TRI_IGETC);
+
+  if (v8connection == nullptr) {
+    TRI_V8_THROW_EXCEPTION_INTERNAL(
+        "startTelemetrics() must be invoked on an arango connection object "
+        "instance.");
+  }
+
+  auto& shellFeature = v8connection->server().getFeature<ShellFeature>();
+
+  shellFeature.startTelemetrics();
+
+  TRI_V8_RETURN_TRUE();
+
+  TRI_V8_TRY_CATCH_END
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief ClientConnection method "restartTelemetrics"
+////////////////////////////////////////////////////////////////////////////////
+
+static void ClientConnection_restartTelemetrics(
+    v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+  if (isExecutionDeadlineReached(isolate)) {
+    return;
+  }
+
+  // get the connection
+  V8ClientConnection* v8connection = TRI_UnwrapClass<V8ClientConnection>(
+      args.Holder(), WRAP_TYPE_CONNECTION, TRI_IGETC);
+
+  if (v8connection == nullptr) {
+    TRI_V8_THROW_EXCEPTION_INTERNAL(
+        "startTelemetrics() must be invoked on an arango connection object "
+        "instance.");
+  }
+
+  auto& shellFeature = v8connection->server().getFeature<ShellFeature>();
+
+  shellFeature.restartTelemetrics();
+
+  TRI_V8_RETURN_TRUE();
+
+  TRI_V8_TRY_CATCH_END
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief ClientConnection method "sendTelemetricsToEndpointTestRedirect"
+////////////////////////////////////////////////////////////////////////////////
+
+static void ClientConnection_sendTelemetricsToEndpointTestRedirect(
```

Comment:
Not sure why this function exists. Can't we just somehow make the original sendTelemetrics stuff use a different URL if a test requires so?